### ISSUE TITLE
Codefix for S3240 can convert to code that has compiler errors

### ISSUE THREAD

**Issue by tvsonar** [SonarSource/sonarlint-vs#230]
```
Name n;
if (condition)
{
  n = new VerySpecialName();
}
else
{
  n = new SpecialName();
}
```
will be converted to 
```
Name n;
n = condition ? new VerySpecialName() : new SpecialName();
```
but because the `whenTrue` is of `VerySpecialName`, the `whenFalse` should also be of the same type.

**Comment by dbolkensteyn** [comment]
IMO this is actually a false positive of the rule
```
n = condition ? new VerySpecialName() : new SpecialName();
``` 

is clearly better than:
```
if (condition)
{
  n = new VerySpecialName();
}
else
{
  n = new SpecialName();
}
```

However, it is much less clear whether or not the if is better than this:
```
n = condition ? (Name)new VerySpecialName() : new SpecialName();
```

Explicit casts are quite ugly, and I can understand that some users will not want to refactor their code as such.

R# also doesn't report at all in this case.
